third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -364,13 +364,33 @@ bool dds_device::impl::init()
 
                     n_stream_options_received++;
 
-                    dds_options options;
-                    for( auto & option : j["options"] )
+                    if( utilities::json::has( j, "options" ) )
                     {
-                        options.push_back( dds_option::from_json( option, stream_it->second->name() ) );
+                        dds_options options;
+                        for( auto & option : j["options"] )
+                        {
+                            options.push_back( dds_option::from_json( option, stream_it->second->name() ) );
+                        }
+
+                        stream_it->second->init_options( options );
                     }
 
-                    stream_it->second->init_options( options );
+                    if( utilities::json::has( j, "intrinsics" ) )
+                    {
+                        auto video_stream = std::dynamic_pointer_cast< dds_video_stream >( stream_it->second );
+                        auto motion_server = std::dynamic_pointer_cast< dds_motion_stream >( stream_it->second );
```

Comment:
it is weird that this param is called motion_server, and one line above, it is called video_stream.
It would seem better if both were stream or server.

Reply:
You are right. Copy-paste leftovers. Both should be named "stream"